### PR TITLE
SNOW-698080: Add functions for user code migration (part 1)

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   SNOW-698080

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   The second part will come with `flatten` related functions.

### CODE REVIEW

**Review by sfc-gh-stan — APPROVED** [review]

File: src/snowflake/snowpark/functions.py [hunk 1]
```diff
@@ -3636,3 +3650,49 @@ def sproc(
             execute_as=execute_as,
             strict=strict,
         )
+
+
+# Add these alias for user code migration
```
[the remark this answers is not in the document]
Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/functions.py:3655]: actually functions after aliases are also for user code migration, which don't exist in Snowflake

File: src/snowflake/snowpark/functions.py [hunk 2]
```diff
@@ -3636,3 +3650,49 @@ def sproc(
...
+    return c.asc()
+
+
+def desc(e: ColumnOrName) -> Column:
+    """Returns a Column expression with values sorted in descending order."""
+    c = _to_col_if_str(e, "desc")
+    return c.desc()
+
+
+def unix_timestamp(e: ColumnOrName, fmt: Optional["Column"] = None) -> Column:
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/functions.py:3682]: I suggest we discuss with more people whether we add these function aliases at the client while Snowflake SQL doesn't have them. An option is to add them to both the client and SQL.

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/functions.py [hunk 3]
```diff
@@ -3668,3 +3690,37 @@ def sproc(
...
+call_builtin = call_function
+builtin = function
+countDistinct = count_distinct
+substr = substring
+to_varchar = to_char
+expr = sql_expr
+date_format = to_date
+first = any_value
+create_map = object_construct
+map_keys = object_keys
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/functions.py:3705]: Snowflake SQL is creating a new data type Map. How about reserve create_may and map_keys for now?

Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/functions.py]: removed

File: src/snowflake/snowpark/functions.py [hunk 4]
```diff
@@ -3668,3 +3690,37 @@ def sproc(
...
+
+# Add these alias for user code migration
+call_builtin = call_function
+builtin = function
+countDistinct = count_distinct
+substr = substring
+to_varchar = to_char
+expr = sql_expr
+date_format = to_date
+first = any_value
```
[the remark this answers is not in the document]
Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/functions.py]: removed

**Review by sfc-gh-yixie — APPROVED** [review]

**Pull request merged**